pipeline-runner/R/gem2s-2-load_user_files.R

```diff
@@ -234,3 +254,84 @@ format_annot <- function(annot_list) {
   rownames(annot) <- annot$input
   return(annot)
 }
+
+# Fix annotations makes annotations compatible between samples with different types.
+# The possible options for features_types at this stage are 0,1,2
+#  0 is SYMBOL/ SYMBOL
+#  1 is IDS/SYMBOL
+#  2 is IDS/IDS
+equalize_annotation_types <- function(annot_list, counts_list, features_types_list, samples) {
+  if (any(features_types_list == 2) && any(features_types_list == 0) && !any(features_types_list == 1)) stop("Incompatible features detected.")
+
+  if (any(features_types_list == 1) && (any(features_types_list == 2) || any(features_types_list == 0))) {
+    annots_with_ids <- unique(do.call("rbind", annot_list[features_types_list == 1]))
+
+    annots_with_ids <- annots_with_ids[!duplicated(annots_with_ids$input), ]
+
+    for (sample in samples) {
```

Comment:
this for looks complex but it might be clearer as:
```
for (sample in samples) {

    if (features_types_list[[sample]] == SYM_SYM) { 

        sample_annot$input <- input_sym_to_ids(annot_list[[sample]])   
        rownames(counts_list[[sample]]) <- sample_annot$input

    } else if (features_types_list[[sample]] == IDS_IDS) {
         sample_annot$name <- name_ids_to_sym(sample_annot)
    }
}
```
so you can read the loop easily and then add a short comment on functions `input_sym_to_ids` and `name_ids_to_sym` to explain what they do

Reply:
You are missing an important if at the beginning, and the important operation of replacing 

annot_list[[sample]] <- sample_annot

Reply:
That being said, I think we can do this for all samples regarding of type, the result will be that for most we will just replace the same thing, and that's what I wanted to avoid. 

It's a matter of clarity vs an extra assignation, but I think in this case the assignation won't be costly so we could potentially replace the for with:

for (sample in samples) { 
     sample_annot <- annot_list[[sample]]
     if (features_types_list[[sample]] == SYM_SYM) { 
          sample_annot$input <- input_sym_to_ids(annot_list[[sample]]) 
          rownames(counts_list[[sample]]) <- sample_annot$input 
 } else if (features_types_list[[sample]] == IDS_IDS) { 
          sample_annot$name <- name_ids_to_sym(sample_annot) 
 } 
      annot_list[[sample]] <- sample_annot
 }